File: Practica4/FiguraGeometrica.py
import math
from multimethod import multimethod

class FiguraGeometrica:
    @multimethod
    def Area(self, radio: float):
        return math.pi * radio * radio

    @multimethod
    def Area(self, base: int, altura: int):
        return base * altura
    @multimethod
    def Area(self, cateto1:float ,cateto2:float):
        return (cateto1*cateto2)/2
    @multimethod
    def Area(self, bmayor:float,bmenor:float,altura:float):
        return (bmayor+bmenor)/2*altura
    @multimethod
    def Area(self, lado:float,apotema:int):
        perimetro=lado*5
        return (perimetro*apotema)/2
f = FiguraGeometrica()
f1 = f.Area(6.3)  # Círculo
f2 = f.Area(12,18)  # Rectángulo
f3 = f.Area(6.0,4.0)#triangulo rectangulo
f4 = f.Area(8.0,6.0,4.0)#trapecio
f5 = f.Area(4.5,4)#pentagono
print("El área del círculo es de:", f1)
print("El área del rectángulo es de:", f2)
print("El area del triangulo rectangulo es de :",f3)
print("El area del trapecio es de :",f4)
print("EL area del pentagono es de :",f5)

File: Practica4/FiguraGeometrica.java
package Figura;   
public class FiguraGeometrica {
    double area(double radio){
        return Math.PI * radio * radio;  
    }
    
    int area(int base, int altura){
        return base * altura;
    }
    
    double area(double cateto1, double cateto2){
        return (cateto1 * cateto2) / 2;
    }
    
    double area(double bmayor, double bmenor, double altura){
        return ((bmayor + bmenor) / 2) * altura;  
    }
    
    double area(double lado, int apotema){
        double p = lado * 5;  
        return (p * apotema) / 2;
    }
    
    public static void main(String[] args) {
        FiguraGeometrica f = new FiguraGeometrica();
        
        System.out.println("Area del Circulo: " + f.area(6.3));
        System.out.println("Area del Rectangulo: " + f.area(12, 18));
        System.out.println("Area del Triangulo rectangulo: " + f.area(6.5, 4.3));
        System.out.println("Area del Trapecio: " + f.area(8.0, 6.0, 4.0));
        System.out.println("Area del Pentagono: " + f.area(4.5, 4));  
    }
}
